test/testers/rust-tester/src/pdre_api/crypto.rs

```diff
@@ -0,0 +1,179 @@
+use super::utils::CryptoApi;
+
+use substrate_primitives::hashing::{twox_128, twox_256, twox_64};
+
+pub fn test_blake2_128() {
+    let mut api = CryptoApi::new();
+
+    let data = String::from("TestInput");
+    let mut output = [0; 16];
+
+    api.rtm_ext_blake2_128(data.as_bytes(), &mut output);
+    assert_eq!(
+        hex::decode("b9f759e8bba7215c55babf022740e70e").unwrap(),
+        output
+    );
+}
+
+pub fn test_blake2_256() {
+    let mut api = CryptoApi::new();
+
+    let data = String::from("TestInput");
+    let mut output = [0; 32];
+
+    api.rtm_ext_blake2_256(data.as_bytes(), &mut output);
+    assert_eq!(
+        hex::decode("205b104b5db516e0a7b61b0243cfc0a7dee470d831e1707ebb99ae6bbd1ca70c").unwrap(),
+        output
+    );
+}
+
+pub fn test_twox_64() {
+    let mut api = CryptoApi::new();
+
+    let data = String::from("TestInput");
+    let mut output = [0; 8];
+
+    api.rtm_ext_twox_64(data.as_bytes(), &mut output);
+    assert_eq!(twox_64(data.as_bytes()), output);
+
+    println!("> testing TwoX 64-bit hash");
```

Comment:
because other implementations need to exactly print the same thing I don't know if it is necessary to print anything beside output of the test. the julia has already given the input information and it has invoked the command line test with its name so it is aware of the input and the functionality being tested.

Reply:
Done!